### PR TITLE
EQL: Allow null tiebreakers inside ordinals/sequences

### LINKED ISSUE #64706: EQL: Allow event in sequences without tiebreakers
Based on the discussion in #64671, the following [inconsistency](https://github.com/elastic/elasticsearch/pull/64671#issuecomment-722984340) has been identified:

When a tiebreaker is missing, we do not validate its existence for event queries however for sequences, we do (and throw a somewhat obscure exception otherwise).
We should align the two - currently a missing tiebreaker occurs after those that specify it. That is null/missing is considered greater than a specified value so that those that do take priority.
We can apply the same for sequences so events with missing tiebreakers instead of throwing an exception will be considered last and incorporated accordingly into the sequence:

`event[2020-11-06T11:22:33.4444][1]` occurs before `event[2020-11-06T11:22:33.4444][null]`

Considering the behavior of event queries, the same ordering characteristics should apply to sequences.
That is:

1. allow a specified tiebreaker with null value. This means no validation of said field, it might be completely missing.
2. if it's missing, null are considered greater than specified values in order to pick those over those that lack this information. This is consistent with event queries.

Supersedes #64690

/cc @astefan @marshallmain @rw-access

### PR DESCRIPTION
**Pull request by costin**
Align Ordinal comparator to consider nulls last (higher) in tiebreakers.
Add unit tests to Ordinal comparisons and criterion extraction.

Fix #64706

### CODE REVIEW

**Comment by costin** [comment]
@elasticmachine update branch

**Comment by costin** [comment]
@elasticmachine run elasticsearch-ci/2

**Comment by costin** [comment]
@elasticmachine update branch

File: x-pack/plugin/eql/src/test/java/org/elasticsearch/xpack/eql/execution/search/CriterionOrdinalExtractionTests.java [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+import static java.util.Collections.emptyList;
+import static java.util.Collections.emptyMap;
+import static java.util.Collections.singletonList;
+
+public class CriterionOrdinalExtractionTests extends ESTestCase {
```
Comment (astefan) [inline x-pack/plugin/eql/src/test/java/org/elasticsearch/xpack/eql/execution/search/CriterionOrdinalExtractionTests.java:28]: I think using "tie" as the short version of "tiebreaker" is confusing and makes the test method names a bit cryptic for someone who is not familiar with the code. "tie" has a meaning and, while it's a short version of "tiebreaker", the reader will not think of "tiebreaker" first.

**Review by astefan — APPROVED** [review]
LGTM. Left one minor comment regarding naming that use "tie" instead of "tiebreaker" in tests.

**Pull request merged**